tests/baselines/reference/overloadresolutionWithConstraintCheckingDeferred.errors.txt

```diff
@@ -53,60 +21,23 @@ overloadresolutionWithConstraintCheckingDeferred.ts(19,14): error TS2344: Type '
     declare function foo(arg: (x: B) => any): number;
     
     var result: number = foo(x => new G(x)); // x has type D, new G(x) fails, so first overload is picked.
-                         ~~~
-!!! error TS2769: No overload matches this call.
-!!! error TS2769:   Overload 1 of 3, '(arg: (x: D) => number): string', gave the following error.
-!!! error TS2769:     Type 'G<A>' is not assignable to type 'number'.
-!!! error TS2769:   Overload 2 of 3, '(arg: (x: C) => any): string', gave the following error.
-!!! error TS2769:     Argument of type '(x: D) => G<A>' is not assignable to parameter of type '(x: C) => any'.
-!!! error TS2769:       Types of parameters 'x' and 'x' are incompatible.
-!!! error TS2769:         Property 'q' is missing in type 'C' but required in type 'D'.
-!!! error TS2769:   Overload 3 of 3, '(arg: (x: B) => any): number', gave the following error.
-!!! error TS2769:     Argument of type '(x: D) => G<A>' is not assignable to parameter of type '(x: B) => any'.
-!!! error TS2769:       Types of parameters 'x' and 'x' are incompatible.
-!!! error TS2769:         Property 'q' is missing in type 'B' but required in type 'D'.
-!!! related TS6502 overloadresolutionWithConstraintCheckingDeferred.ts:10:27: The expected type comes from the return type of this signature.
-!!! related TS2728 overloadresolutionWithConstraintCheckingDeferred.ts:4:15: 'q' is declared here.
-!!! related TS2728 overloadresolutionWithConstraintCheckingDeferred.ts:4:15: 'q' is declared here.
+        ~~~~~~
+!!! error TS2322: Type 'string' is not assignable to type 'number'.

```

Comment:
Previously `x` would have the type from the first overload, since it'd get immediately fixed, now `x` gets the type form the first overload that passes, which is the _second_, since `G` is assignable to `any` and not `number`. So even though the call to `G` fails with `x` being C`, the call as a whole still works out. This is actually a place we could improve - rather than just checking the overload's validity, we could check for errors in the whole subexpression tree - but that's future work.